Diagram/AutoLayoutSettings.cs

```diff
@@ -989,16 +1003,19 @@ private void MSAGL(List<NodeBase> all_nodes, List<LinkBase> all_links)
             {
                 node.MoveTo(gnode.Center.X - gnode.Width / 2 - graph.Left, gnode.Center.Y - gnode.Height / 2 - graph.Bottom);
             }
-            foreach (var (link, glink) in all_links.Zip(graph.Edges, (a, b) => (a, b)))
+            if (!PreservePorts)
             {
-                if (link.Source.Node != null && link.Target.Node != null)

```

Comment:
Why does this line need to be removed?

Reply:
It's not removed, just "moved" (not really - displayed a bit misleadingly due to diff)